include/aws/s3/private/s3_client_impl.h

```diff
@@ -15,7 +15,9 @@
 #include <aws/common/mutex.h>
 #include <aws/common/ref_count.h>
 #include <aws/common/task_scheduler.h>
+#include <aws/http/connection.h>
 #include <aws/http/connection_manager.h>
+#include <aws/http/proxy.h>
```

Comment:
Prefer to use forward-declarations instead of adding includes, whenever you can get away with it. This keeps down compile times in large C/C++ projects.

Reply:
I have tried that. The problem with that is that we can't write tests for these variables that verify that allocation, deallocation, and assignment are working fine. As it is a private header, it won't be included in dependencies anyhow.